internal/prompter/prompter.go

```diff
@@ -0,0 +1,135 @@
+package prompter
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/AlecAivazis/survey/v2/terminal"
+	"github.com/cli/cli/v2/internal/ghinstance"
+	"github.com/cli/cli/v2/pkg/surveyext"
+)
+
+//go:generate moq -rm -out prompter_mock.go . Prompter
+type Prompter interface {
+	Select(string, string, []string) (int, error)
+	MultiSelect(string, string, []string) (int, error)
+	Input(string, string) (string, error)
+	InputHostname() (string, error)
+	Password(string) (string, error)
+	AuthToken() (string, error)
+	Confirm(string, bool) (bool, error)
+	MarkdownEditor(string, string, bool) (string, error)
```

Comment:
Thinking a bit to the future I wonder if we would be better served making this a generic `Editor` function that takes in a filetype/filename as an argument rather than hardcoding `*.md` in the implementation below. My thought would be that once we are able to add this to `go-gh` I can see a generic `Editor` function providing more value to extension authors even though `gh` only uses it for markdown.

Reply:
Along these same lines I am wondering if `InputHostname` and `AuthToken` are really necessary since they are specific cases of `Input` and `Password` respectively that may only apply to `gh`.

Reply:
I discussed this with Mislav and we agreed that more, very specific prompts was preferable to prompting functions that took a bunch of options. I'd like to go with what's here for now but if in practice it sucks I'm ok revisiting before exposing in go-gh.